src/controllers/HistoryController.ts

```diff
@@ -89,16 +89,19 @@ export class HistoryController extends RootComponent {
     this.logger.trace('Update history hash');
 
     const hash = '#' + this.hashUtils.encodeValues(values);
-    this.ignoreNextHashChange = this.window.location.hash != hash;
+    const hashHasChanged = this.window.location.hash != hash;
+    this.ignoreNextHashChange = hashHasChanged;
 
     this.logger.trace('ignoreNextHashChange', this.ignoreNextHashChange);
     this.logger.trace('initialHashChange', this.initialHashChange);
     this.logger.trace('from', this.window.location.hash, 'to', hash);
 
     if (this.initialHashChange) {
       this.initialHashChange = false;
-      this.window.location.replace(hash);
-      this.logger.trace('History hash modified', hash);
+      if (hashHasChanged) {
```

Comment:
Is the `hashHasChanged` variable needed or would it be possible to use `this.ignoreNextHashChange`?

Reply:
I did this to make it clear what the intent is.